avfilter/vf_xcam: add xcam video filter

zongwave opened the pull request:
configure option: --enable-libxcam

xcam filter is a wrapper of libxcam project which provides rich image processing
and video analytic functions such as: 360 video stitching (support 2/3/4 camera inputs),
surround view stitching, wavelet denoise, 3D denoise, Wide dynamic range (WDR) and
digital video stabilization, etc.

libxcam library optimized algorithms by AVX, GLES and Vulkan depends on
hardware configuration to get good performance.

libxcam is hosted at https://github.com/intel/libxcam

360 video stitch command line:
$ffmpeg -i input0.mp4 -i input1.mp4 -i input2.mp4 -filter_complex "xcam=inputs=3 \
:name=stitch:w=7680:h=3840:fmt=nv12:params=help=1 module=soft cammodel=camc3c8k \
fisheyenum=3 levels=1 dewarp=sphere scale=dualconst fm=cluster fmframes=120 \
fmstatus=fmfirst scopic=stereoleft" output.mp4

Reviewed-by: Zong Wei <wei.zong@intel.com>

uartie commented:
Please fix `configure` file permissions:
`/bin/sh: 1: exec: ./configure: Permission denied`

liuyinhangx committed:
avfilter/vf_xcam: add xcam video filter

xcam filter is a wrapper of libxcam project which supports 360 video stitching,
automotive surround view stitching, digital video stabilization (DVS),
Wide dynamic range (WDR), wavelet noise reduction, 3D noise reduction, etc.

libxcam library optimized algorithms by AVX, GLES and Vulkan depends on
hardware configuration to get good performance.

libxcam project is hosted at https://github.com/intel/libxcam
please refer to https://github.com/intel/libxcam/wiki/Build to build libxcam library.

To enable xcam video filter, configure FFmpeg with option
--enable-libxcam.

Here are the features provided by xcam video filter:
- stitch     CPU|GLES|Vulkan stitching
- stitchcl   OpenCL stitching
- fisheye    Fisheye calibration
- 3dnr       3D denoising
- waveletnr  Wavelet denoising
- dvs        Digital video stabilization
- defog      Fog removal

Use 'ffmpeg -h filter=xcam' to get the common parameters, stitch and
stitchcl have private parameters, use 'params=help=1' to get the private
parameters. For more detailed test cases posted at:
https://github.com/intel/libxcam/wiki/Tests#1-ffmpeg-xcam.

Reviewed-by: Zong Wei <wei.zong@intel.com>
